src/read_utils.rs

```diff
@@ -189,23 +190,47 @@ where
 }
 
 /// Read a vector of objects from a zip in a file_handler
-pub(crate) fn read_objects<H, O>(file_handler: &mut H, file_name: &str) -> Result<Vec<O>>
```

Comment:
Most of `read_objects()` and `read_objects_skip_error()` are identical. Can they be factorized?